### ISSUE TITLE
[Idea] Conditional Animations

### ISSUE THREAD

**Issue by ScottHerrmann** [WeakAuras/WeakAuras2#314]
It would be interesting to have an optional condition for running an animation. Currently (as far as I can tell), it's not possible to start animations via triggers.

**Comment by InfusOnWoW** [comment]
Yep, that's currently not possible. I do have plans to eventually add that, it'll be a while though.

**Comment by ScottHerrmann** [comment]
I found a workaround for now. I set an aura_env variable in the trigger updater for when a trigger is valid and enclose the animation function in an if statement based on that variable. It works for now.

**Comment by Arxareon** [comment]
I hope this is still on the table! I'd very much like to see an implementation of this.
Maybe: Condition Then: Toggle Main Animation. If there is an animation set, turn it on and off based on the condition.

**Comment by khamarr3524** [comment]
> Out of question; how did you manage to do that Scott. I've been looking at trying to animate pulse or 'blinking' when the target debuff is ready. (I'm currently doing this by glow for the moment).
> 
> I've not done any Lua coding, but I may start to look into it. The biggest hurdle for me at the moment is the syntax of the language.

I somewhat documented this in my second post. I don't think anything was ever done to get this implemented. I haven't been doing as much weakauras stuff as before. If you still have a question or if this is still needed/if I end up needing it, I'll go in and implement the changes and do a PR myself and will update this accordingly. Sorry for my absence!

**Comment by Ernest314** [comment]
There recently was a short conversation in the Discord about weakaura packs with an excessive number of individual weakauras (in the hundreds), instead of properly using conditionals. This feature has the potential to allow those packs to be greatly simplified, while keeping the simple & accessible preset animations currently available.

Some example use cases that currently require custom code:
- fading out/in a weakaura when entering/exiting a vehicle
- gradual (fade in) color change at certain % energy
- pulse weakaura when capping spell charges

**Comment by eranikid** [comment]
Another use case, to add to the above, is this - I want to have my ability cooldown auras to:
- have 50% alpha when on cooldown
- pulse once when coming off cooldown

Currently I have found two ways to do this, both flawed:

1. Create 2 WAs. One is active when the ability is on CD, another one is active when the ability is ready. First one has constant 50% alpha, second one has start animation. The flaw is that I have to manage 2 auras instead of one. Partly mitigated by putting both auras into a group, but I want these auras in a dynamic group, and [groups cannot be placed inside dynamic groups](https://github.com/WeakAuras/WeakAuras2/issues/3786)
2. Create 1 WA, attach custom code to conditions:
```lua
-- On cooldown: false
aura_env.ready_at = GetTime()
aura_env.on_cooldown = false

-- On cooldown: true
aura_env.on_cooldown = true
```
And handcode custom _Main_ animation, which only does something when `GetTime() - aura_env.ready_at` is less than some threshold. The flaw is having to maintain custom code. Given I am terrible at WA crafting, the flaw is even more severe, as I now have to set alpha, animation speed and parameters all in code, instead of using conventional UI controls.

@InfusOnWoW maybe there is a way to trigger start animation in custom function of condition? In future it would be nice to have access to condition property change "Play start/stop animation once"
